Gems/ScriptCanvasDeveloper/Code/Editor/Source/Tools/TranslationBrowser/TranslationBrowser.cpp

```diff
@@ -0,0 +1,897 @@
+
+
+                            const AZStd::string parameterName = resultParameter->m_name;
+                            GraphCanvas::TranslationKeyedString oldArgName(parameterName, translationContext, oldKey);
+                            GraphCanvas::TranslationKeyedString oldArgTooltip(resultDescription, translationContext, oldTooltipKey);
+
+
+                            result.m_typeId = resultKey;
+                            result.m_details.m_name = oldArgName.GetDisplayString();
+                            result.m_details.m_tooltip = oldArgTooltip.GetDisplayString();
+
+                            methodEntry.m_results.push_back(result);
+                        }
+
+                        entry.m_methods.push_back(methodEntry);
+                    }
+                }
+
+                translationRoot.m_entries.push_back(entry);
+            }
+
+            AZStd::string content = MakeJSON(translationRoot);
+            m_ui->sourceTranslationData->setPlainText(content.c_str());
+        }
+    }
+
+    void TranslationBrowser::OnDoubleClick()
+    {
+
+    }
+
+    void TranslationBrowser::OnSystemTick()
+    {
+    }
+
+
+
+    void TranslationBrowser::closeEvent(QCloseEvent* event)
+    {
+
+        AzQtComponents::StyledDialog::closeEvent(event);
+    }
+
+    void TranslationBrowser::LoadJSONForClass(const AZStd::string& className)
+    {
+      
+        AZStd::string fileName = AZStd::string::format("%s.names", className.c_str());
+        fileName = GraphCanvas::TranslationKey::Sanitize(fileName);
+        AZStd::to_lower(fileName.begin(), fileName.end());
+
+        AZStd::string found;
+
+        // Find any TranslationAsset files that may have translation database key/values
+        AZ::Data::AssetCatalogRequestBus::Broadcast(&AZ::Data::AssetCatalogRequestBus::Events::EnumerateAssets,
+            nullptr,
+            [fileName, &found](const AZ::Data::AssetId /*assetId*/, const AZ::Data::AssetInfo& assetInfo) {
+                const auto assetType = azrtti_typeid<GraphCanvas::TranslationAsset>();
+                if (found.empty() && assetInfo.m_assetType == assetType)
+                {
+                    AZStd::string testFile;
+                    AZ::StringFunc::Path::GetFullFileName(assetInfo.m_relativePath.c_str(), testFile);
+                    testFile = GraphCanvas::TranslationKey::Sanitize(testFile);
+                    AZStd::to_lower(testFile.begin(), testFile.end());
+
+                    if (fileName == testFile)
+                    {
+                        found = assetInfo.m_relativePath;
+                    }
+                }
+            },
+            []() {
+                
+            }
+            );
+
+
+        if (!found.empty() && AZ::IO::FileIOBase::GetInstance()->Exists(found.c_str()))
+        {
+            m_ui->btnOpenInExplorer->setEnabled(true);
+            m_selection = found;
+
+            AZ::IO::FileIOBase* fileIO = AZ::IO::FileIOBase::GetInstance();
```

Comment:
The fileIO could be cached into a local variable before line 654. The Exists check on line 649 queries the FileIOBase Instance

Reply:
Alternatively you can just use an AZ::IO::FileReader.
It will simplify the FileClose logic, since it acts as an RAII handle.

For example
```
FileReader fileHandle(found.c_str(), AZ::IO::FileIOBase::GetInstance());
```

Reply:
Since this is likely going away, I'll do the FileIO cache version